src/hooks/useAsyncFetch.js

```diff
@@ -0,0 +1,18 @@
+import { useProducts } from "../context/product-context";
+import {useEffect} from "react";
+import axios from "axios";
+
+export function useAsyncFetch({url, dispatchType, dispatchPayload}){
+    const {dispatch}  = useProducts();
+    useEffect(()=>{
+        (async function asyncProductFetch(){
+            try{
+                const {data} = await axios.get(url);
+                console.log("Calling api"+url);
```

Comment:
can remove the unused console.log()s from the code.

Reply:
I have added it intentionally to keep a watch on the number of API calls being made, so will try to reduce the number of API calls if found something is redundant. After the app is complete I will remove the console.log(), for now I will keep it.